iroh-p2p/src/behaviour/peer_manager.rs

```diff
@@ -186,8 +192,26 @@ impl NetworkBehaviour for PeerManager {
     fn poll(
         &mut self,
         _cx: &mut Context<'_>,
-        _params: &mut impl PollParameters,
+        params: &mut impl PollParameters,
     ) -> Poll<NetworkBehaviourAction<Self::OutEvent, Self::ConnectionHandler>> {
+        // TODO(ramfox):
+        // We can only get the supported protocols of the local node by examining the
+        // `PollParameters`, which mean you can only get the supported protocols by examining the
+        // `PollParameters` in this method (`poll`) of a network behaviour.
+        // I injected this responsibility in the `peer_manager`, because it's the only "simple"
+        // network behaviour we have implemented.
+        // There is an issue up to remove `PollParameters`, and a discussion into how to instead
+        // get the `supported_protocols` of the node:
+        // https://github.com/libp2p/rust-libp2p/issues/3124
+        // When that is resolved, we can hopefully remove this responsibility from the `peer_manager`,
+        // where it, frankly, doesn't belong.
+        if self.supported_protocols.is_empty() {
+            self.supported_protocols = params
+                .supported_protocols()
+                .map(|p| String::from_utf8_lossy(&p).to_string())
+                .collect();
+        }
+
```

Comment:
@dignifiedquire would specifically like your eyes on this.

I gave some rationale as to why this is how I decided to proceed, but I'm aware this is an inelegant solution to an inelegant problem.

Reply:
seems fine to me, we can just update this code once libp2p provides a better method